File: ios/Sasquatsh/Views/Shared/GroupCard.swift
import SwiftUI

struct GroupCard: View {
    let group: GroupSummary

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                if let logoUrl = group.logoUrl, let url = URL(string: logoUrl) {
                    AsyncImage(url: url) { image in
                        image.resizable().aspectRatio(contentMode: .fill)
                    } placeholder: {
                        groupIcon
                    }
                    .frame(width: 44, height: 44)
                    .clipShape(RoundedRectangle(cornerRadius: MD3Shape.small))
                } else {
                    groupIcon
                }

                VStack(alignment: .leading, spacing: 2) {
                    Text(group.name)
                        .font(.md3TitleMedium)
                        .foregroundStyle(Color.md3OnSurface)
                        .lineLimit(1)

                    if let location = locationText {
                        Text(location)
                            .font(.md3BodySmall)
                            .foregroundStyle(Color.md3OnSurfaceVariant)
                    }
                }

                Spacer()

                if let role = group.userRole {
                    BadgeView(text: role.displayName, color: .md3PrimaryContainer)
                }
            }

            if let description = group.description, !description.isEmpty {
                Text(description)
                    .font(.md3BodyMedium)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
                    .lineLimit(2)
            }

            HStack(spacing: 16) {
                Label {
                    Text("\(group.memberCount) members")
                        .font(.md3BodySmall)
                } icon: {
                    Image(systemName: "person.3")
                        .foregroundStyle(Color.md3Primary)
                }

                BadgeView(text: group.groupType.displayName, color: .md3TertiaryContainer)

                Spacer()

                Image(systemName: policyIcon)
                    .font(.md3BodySmall)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
                Text(group.joinPolicy.displayName)
                    .font(.md3BodySmall)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
            }
        }
        .padding()
        .cardStyle()
    }

    private var groupIcon: some View {
        RoundedRectangle(cornerRadius: MD3Shape.small)
            .fill(Color.md3PrimaryContainer)
            .frame(width: 44, height: 44)
            .overlay {
                Image(systemName: "person.3.fill")
                    .foregroundStyle(Color.md3OnPrimaryContainer)
            }
    }

    private var locationText: String? {
        if let city = group.locationCity, let state = group.locationState {
            return "\(city), \(state)"
        }
        return nil
    }

    private var policyIcon: String {
        switch group.joinPolicy {
        case .open: return "door.left.hand.open"
        case .request: return "hand.raised"
        case .invite_only: return "lock"
        }
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/shared/GroupCard.kt
package com.sasquatsh.app.views.shared

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Groups
import androidx.compose.material.icons.outlined.DoorFront
import androidx.compose.material.icons.outlined.Groups
import androidx.compose.material.icons.outlined.Lock
import androidx.compose.material.icons.outlined.PanTool
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import coil.compose.AsyncImage
import com.sasquatsh.app.models.GroupSummary
import com.sasquatsh.app.models.JoinPolicy

@Composable
fun GroupCard(
    group: GroupSummary,
    modifier: Modifier = Modifier,
    onClick: (() -> Unit)? = null
) {
    Card(
        onClick = { onClick?.invoke() },
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceContainer
        ),
        shape = RoundedCornerShape(12.dp),
        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp)
    ) {
        Column(
            verticalArrangement = Arrangement.spacedBy(8.dp),
            modifier = Modifier.padding(16.dp)
        ) {
            // Header: logo, name, location, role badge
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                val logoUrl = group.logoUrl
                if (logoUrl != null) {
                    AsyncImage(
                        model = logoUrl,
                        contentDescription = group.name,
                        contentScale = ContentScale.Crop,
                        modifier = Modifier
                            .size(44.dp)
                            .clip(RoundedCornerShape(8.dp))
                            .background(MaterialTheme.colorScheme.primaryContainer)
                    )
                } else {
                    GroupIconPlaceholder()
                }

                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        text = group.name,
                        style = MaterialTheme.typography.titleMedium,
                        color = MaterialTheme.colorScheme.onSurface,
                        maxLines = 1,
                        overflow = TextOverflow.Ellipsis
                    )
                    val locationText = buildLocationText(group.locationCity, group.locationState)
                    if (locationText != null) {
                        Text(
                            text = locationText,
                            style = MaterialTheme.typography.bodySmall,
                            color = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }

                val role = group.userRole
                if (role != null) {
                    BadgeView(
                        text = role.displayName,
                        color = MaterialTheme.colorScheme.primaryContainer
                    )
                }
            }

            // Description
            val description = group.description
            if (!description.isNullOrEmpty()) {
                Text(
                    text = description,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis
                )
            }

            // Footer: member count, group type, join policy
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(16.dp),
                modifier = Modifier.fillMaxWidth()
            ) {
                // Member count
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(4.dp)
                ) {
                    Icon(
                        imageVector = Icons.Outlined.Groups,
                        contentDescription = null,
                        tint = MaterialTheme.colorScheme.primary,
                        modifier = Modifier.size(16.dp)
                    )
                    Text(
                        text = "${group.memberCount} members",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }

                // Group type badge
                BadgeView(
                    text = group.groupType.displayName,
                    color = MaterialTheme.colorScheme.tertiaryContainer
                )

                Spacer(modifier = Modifier.weight(1f))

                // Join policy
                Row(
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(4.dp)
                ) {
                    Icon(
                        imageVector = policyIcon(group.joinPolicy),
                        contentDescription = null,
                        tint = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.size(14.dp)
                    )
                    Text(
                        text = group.joinPolicy.displayName,
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }
        }
    }
}

@Composable
private fun GroupIconPlaceholder() {
    Box(
        modifier = Modifier
            .size(44.dp)
            .clip(RoundedCornerShape(8.dp))
            .background(MaterialTheme.colorScheme.primaryContainer),
        contentAlignment = Alignment.Center
    ) {
        Icon(
            imageVector = Icons.Filled.Groups,
            contentDescription = null,
            tint = MaterialTheme.colorScheme.onPrimaryContainer,
            modifier = Modifier.size(24.dp)
        )
    }
}

private fun buildLocationText(city: String?, state: String?): String? {
    return if (city != null && state != null) "$city, $state" else null
}

private fun policyIcon(policy: JoinPolicy) = when (policy) {
    JoinPolicy.OPEN -> Icons.Outlined.DoorFront
    JoinPolicy.REQUEST -> Icons.Outlined.PanTool
    JoinPolicy.INVITE_ONLY -> Icons.Outlined.Lock
}
